### PR TITLE
GLT-2181: support multiple simultaneous runs per sequencer

### PR DESCRIPTION
**Pull request by djcooke**
Won't merge until runscanner version is changed to a release

### CODE REVIEW

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/InstrumentPositionStatus.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+import javax.persistence.Table;
+
+import org.hibernate.annotations.Immutable;
+
+import uk.ac.bbsrc.tgac.miso.core.data.InstrumentPositionStatus.InstrumentPositionStatusId;
+import uk.ac.bbsrc.tgac.miso.core.data.impl.InstrumentImpl;
+
+@Entity
+@Immutable
+@Table(name = "InstrumentStats")
```
[the remark this answers is not in the document]
Comment (djcooke) [reply core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/InstrumentPositionStatus.java]: 🤷‍♂️ I didn't name it 😆

[the remark this answers is not in the document]
Comment (callunity) [reply core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/InstrumentPositionStatus.java]: Given that we might plausibly want to add an `InstrumentStats` table in the future, could it be renamed?

File: sqlstore/src/main/resources/db/migration/V8504__sequencer_positions.sql [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+UPDATE Run_SequencerPartitionContainer rspc
+JOIN Run r ON r.runId = rspc.run_runId
+JOIN Instrument inst ON inst.instrumentId = r.instrumentId
+JOIN Platform p ON p.platformId = inst.platformId
+SET rspc.positionId = (
+  SELECT positionId FROM PlatformPosition
+  WHERE platformId = inst.platformId
+  AND alias = 'B'
+)
+WHERE (p.instrumentModel LIKE 'Illumina HiSeq%' OR p.instrumentModel LIKE 'Illumina NovaSeq%')
```
Comment (callunity) [inline sqlstore/src/main/resources/db/migration/V8504__sequencer_positions.sql]: Does the NextSeq not have A & B sides?

Comment (djcooke) [reply sqlstore/src/main/resources/db/migration/V8504__sequencer_positions.sql]: It doesn't seem to. All NextSeq run names contain an "A" where the HiSeq/NovaSeq would indicate the position, and the RunParameters.xml don't indicate a side at all.

File: miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultServiceRecordService.java [hunk 3]
```diff
@@ -70,6 +82,24 @@ private void applyRecordChanges(ServiceRecord target, ServiceRecord source) {
     target.setEndTime(source.getEndTime());
   }
 
+  private void ValidateChange(ServiceRecord record, ServiceRecord beforeChange) throws IOException {
```
[the remark this answers is not in the document]
Comment (djcooke) [reply miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultServiceRecordService.java]: fixed

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/InstrumentPositionStatus.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (djcooke) [reply core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/InstrumentPositionStatus.java]: Renamed `InstrumentPositionStatus` to match class name

**Pull request merged**